smart-contracts/contracts/CosmosBridge.sol

```diff
@@ -232,48 +336,30 @@ contract CosmosBridge is CosmosBridgeStorage, Oracle {
 
         // ensure the networkDescriptor matches
         if (!claimData.doublePeg) {
-            require(verifyNetworkDescriptor(claimData.networkDescriptor), "INV_NET_DESC");
+            require(_verifyNetworkDescriptor(claimData.networkDescriptor), "INV_NET_DESC");
         }
         
-        // ensure there are no duplicate signers
-        require(
-            !findDup(signatureData), "DUP_SIGNER"
-        );
+        (bool dup, uint256 pow) = getSignedPowerAndFindDup(signatureData, hashDigest);
```

Comment:
Let's remove the `dup` variable and just add a require statement inside of the `getSignedPowerAndFindDup` function.